plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -623,4 +623,14 @@ private static void validateOrcBloomFilterColumns(ConnectorTableMetadata tableMe
             throw new TrinoException(INVALID_TABLE_PROPERTY, format("Orc bloom filter columns %s not present in schema", Sets.difference(ImmutableSet.copyOf(orcBloomFilterColumns), allColumns)));
         }
     }
+
+    public static String removeRedundantSlash(String location)
+    {
+        // Version 393 and 394 stored metadata location with double slash https://github.com/trinodb/trino/pull/13530
+        // e.g. s3://bucket/db/table//metadata/00001.metadata.json
+        // It caused failure when accessing S3 objects https://github.com/trinodb/trino/issues/14299
+        // Version 395 fixed the above issue by removing trailing slash https://github.com/trinodb/trino/pull/13984,
+        // but the change was insufficient for existing table cases created by 393 and 394. This method covers existing table cases.
+        return location.replace("//metadata/", "/metadata/");
```

Comment:
is it possible that the metadata path really contains two slashes?
for example, what if table was created with `location = 's3://bucket/my_table/////'`?

also, is it possible that `//metadata/` substring occurs somewhere else within the location?
for example, what if my bucket is called `metadata`?
or, i have a table created with `location = 's3://bucket/i/love/slashes/and//////metadata/my_schema/my_table'`?